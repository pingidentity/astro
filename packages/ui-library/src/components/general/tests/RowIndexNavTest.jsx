window.__DEV__ = true;

jest.dontMock("../RowIndexNav");

describe("Indent", function () {
    var React = require("react"),
        ReactTestUtils = require("react-dom/test-utils"),
        TestUtils = require("../../../testutil/TestUtils"),
        RowIndexNav = require("../RowIndexNav"),
        _ = require("underscore");

    var activeIndexes = "ABCDEZ".split(""),
        component,
        componentId = "index-nav",
        indexes = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split(""),
        navItems,
        onClickFunction = jest.genMockFunction(),
        selectedClass = "selected",
        selectedIndex = 2;


    function getComponent (opts) {
        opts = _.defaults(opts || {}, {
            "data-id": componentId,
            indexes: indexes,
            onClick: onClickFunction,
            activeIndexes: activeIndexes,
            selectedIndex: activeIndexes[selectedIndex]
        });
        return ReactTestUtils.renderIntoDocument(<RowIndexNav {...opts} />);
    }


    beforeEach(function () {
        component = getComponent();
        navItems = TestUtils.scryRenderedDOMNodesWithTag(component, "li");
    });

    it("renders the component", function () {
        var container = ReactTestUtils.findRenderedDOMComponentWithClass(component, "row-index-nav");

        expect(ReactTestUtils.isDOMComponent(component)).toBeDefined();
        expect(container).toBeTruthy();
        expect(navItems.length).toEqual(indexes.length);
    });

    it("renders the nav items with the proper content and css classes", function () {
        var disabledClass = "disabled",
            testIndexActive = 3,
            testIndexInactive = 5;

        expect(navItems[testIndexActive].textContent).toEqual(indexes[testIndexActive]);
        expect(navItems[testIndexInactive].textContent).toEqual(indexes[testIndexInactive]);

        expect(navItems[testIndexActive].getAttribute("class")).not.toContain(disabledClass);
        expect(navItems[testIndexInactive].getAttribute("class")).toContain(disabledClass);

        expect(navItems[testIndexActive].getAttribute("class")).not.toContain(selectedClass);
        expect(navItems[testIndexInactive].getAttribute("class")).not.toContain(selectedClass);
        expect(navItems[selectedIndex].getAttribute("class")).toContain(selectedClass);
    });

    it("calls onClick callback", function () {
        var testIndexActive = 3,
            testIndexInactive = 5;

        ReactTestUtils.Simulate.click(navItems[testIndexInactive]);
        expect(component.props.onClick).not.toBeCalled();

        ReactTestUtils.Simulate.click(navItems[testIndexActive]);
        expect(component.props.onClick).toBeCalled();
    });
});
