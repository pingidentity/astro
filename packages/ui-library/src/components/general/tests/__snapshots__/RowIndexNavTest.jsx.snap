// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indent data-id's don't change 1`] = `
Array [
  "row-index-nav",
  "row-index-nav-items",
  "nav-item_A",
  "nav-item_B",
  "nav-item_C",
  "nav-item_D",
  "nav-item_E",
  "nav-item_F",
  "nav-item_G",
  "nav-item_H",
  "nav-item_I",
  "nav-item_J",
  "nav-item_K",
  "nav-item_L",
  "nav-item_M",
  "nav-item_N",
  "nav-item_O",
  "nav-item_P",
  "nav-item_Q",
  "nav-item_R",
  "nav-item_S",
  "nav-item_T",
  "nav-item_U",
  "nav-item_V",
  "nav-item_W",
  "nav-item_X",
  "nav-item_Y",
  "nav-item_Z",
]
`;
